### PR TITLE
samples: add sample code to query regional Dialogflow CX agent

### PR DESCRIPTION
**Pull request by jessiexjiang27**
Add logic to query regional Dialogflow CX endpoint for the following methods:
- CreateFlow
- CreateIntent
- CreatePage
- DetectIntent
- DetectIntentStream

### CODE REVIEW

File: samples/snippets/src/main/java/dialogflow/cx/CreateFlow.java [hunk 1]
```diff
@@ -41,8 +42,16 @@ public static Flow createFlow(
       String agentId,
       Map<String, String> eventsToFulfillmentMessages)
       throws IOException, ApiException {
+    FlowsSettings.Builder flowsSettingsBuilder = FlowsSettings.newBuilder();
+    if (locationId == "global") {
```
Comment (chingor13) [inline samples/snippets/src/main/java/dialogflow/cx/CreateFlow.java:46]: You probably want `.equals()` here.

File: samples/snippets/src/main/java/dialogflow/cx/CreateIntent.java [hunk 2]
```diff
@@ -38,8 +39,16 @@ public static Intent createIntent(
       String agentId,
       List<String> trainingPhrasesParts)
       throws IOException, ApiException {
+    IntentsSettings.Builder intentsSettingsBuilder = IntentsSettings.newBuilder();
+    if (locationId == "global") {
```
Comment (chingor13) [inline samples/snippets/src/main/java/dialogflow/cx/CreateIntent.java:43]: `.equals()`

File: samples/snippets/src/main/java/dialogflow/cx/CreatePage.java [hunk 3]
```diff
@@ -42,8 +43,16 @@ public static Page createPage(
       String flowId,
       List<String> entryTexts)
       throws IOException, ApiException {
+    PagesSettings.Builder pagesSettingsBuilder = PagesSettings.newBuilder();
+    if (locationId == "global") {
```
Comment (chingor13) [inline samples/snippets/src/main/java/dialogflow/cx/CreatePage.java:47]: `.equals()`

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntent.java [hunk 4]
```diff
@@ -42,9 +43,17 @@
       List<String> texts,
       String languageCode)
       throws IOException, ApiException {
+    SessionsSettings.Builder sessionsSettingsBuilder = SessionsSettings.newBuilder();
+    if (locationId == "global") {
```
Comment (chingor13) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntent.java:47]: `.equals()`

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentStream.java [hunk 5]
```diff
@@ -39,8 +40,16 @@
   public static void detectIntentStream(
       String projectId, String locationId, String agentId, String sessionId, String audioFilePath)
       throws ApiException, IOException {
+    SessionsSettings.Builder sessionsSettingsBuilder = SessionsSettings.newBuilder();
+    if (locationId == "global") {
```
Comment (chingor13) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentStream.java:44]: `.equals()`

**Review by chingor13 — CHANGES_REQUESTED** [review]
Overall, do we want to have separate samples for regional endpoints versus the global (or default) endpoint?

Is the documentation where these are referenced ok with a sample with the conditional logic versus suggesting separate snippets for using regional endpoints?

File: samples/snippets/src/main/java/dialogflow/cx/CreateFlow.java (hunk 1 above)
Comment (jessiexjiang27) [reply samples/snippets/src/main/java/dialogflow/cx/CreateFlow.java]: Thank you!

**Comment by jessiexjiang27** [comment]
> Overall, do we want to have separate samples for regional endpoints versus the global (or default) endpoint?
> 
> Is the documentation where these are referenced ok with a sample with the conditional logic versus suggesting separate snippets for using regional endpoints?

I think for CX it's better to have samples in one place with the condition, because, different from DF v2, location ID must always be present in the resource names, so the only difference is setting different endpoints based on location ID, which may be better to put in one sample.

The documentation should be ok to reference the current snippet.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**